### DISCUSSION TITLE
Add initial delay at acquisition start

### DISCUSSION THREAD

**Discussion by lachioma** [bonsai-rx/bonsai#1431] [Q&A] [answered]
Hi all
I would like the image acquisition to start a certain interval of time after the audio capture (e.g. 1 s). 
I tried in various ways adding a Timer node at the beginning of the video capture chain but I didn't manage.
Can anyone please help?

Many thanks!
-ale

![image](https://github.com/bonsai-rx/bonsai/assets/29898879/92583392-cf10-4ad5-b156-d378397dd662)

**Comment by bruno-f-cruz** [comment, marked as answer]
Depends exactly what you want you mean by image acquisition. Is it the logging or the start of the camera?
if it is the logging you can do:

![image](https://github.com/bonsai-rx/bonsai/assets/7049351/d537bdbf-e17d-4f70-be71-1bf59310c1cf)

If it is the start of the camera:

![image](https://github.com/bonsai-rx/bonsai/assets/7049351/0ee34e6b-15b0-475c-be0b-af037d02f4d9)


Also, if you want to delay 1 second AFTER the first audio buffer:
![image](https://github.com/bonsai-rx/bonsai/assets/7049351/3907ef3b-3eb3-4f87-b10e-bc9fa6652614)

**Reply by lachioma** [reply]
Thanks Bruno.
I want to delay 1 second the acquisition+logging of the camera after the audio buffer, so it's the last solution of the 3 you gave.

**Comment by lachioma** [comment]
Now, I would like (1) the audio recording to last a certain amount of time, e.g. 100 s, and (2) the camera capture to terminate 1 second before the audio recording. 
How could I do that?

**Reply by bruno-f-cruz** [reply]
Take a look at [`TakeUntil`](https://bonsai-rx.org/docs/api/Bonsai.Reactive.TakeUntil.html) or [`TakeWhile`](https://bonsai-rx.org/docs/api/Bonsai.Reactive.TakeWhile.html).

Here's an example:
https://github.com/orgs/bonsai-rx/discussions/1023#discussioncomment-3556108

**Reply by lachioma** [reply]
Thank you Bruno, I ended up using `TakeUntil`.
